control/available_plugin.go

```diff
@@ -440,6 +456,59 @@ func (ap *availablePlugins) collectMetrics(pluginKey string, metricTypes []core.
 	return results, nil
 }
 
+func (ap *availablePlugins) streamMetrics(
+	pluginKey string,
+	metricTypes []core.Metric,
+	taskID string,
+	maxCollectDuration time.Duration,
+	maxMetricsBuffer int64) (chan []core.Metric, chan error, error) {
+
+	pool, serr := ap.getPool(pluginKey)
+	if serr != nil {
+		return nil, nil, serr
+	}
+	if pool == nil {
+		return nil, nil, serror.New(ErrPoolNotFound, map[string]interface{}{"pool-key": pluginKey})
+	}
+
+	if pool.Strategy() == nil {
+		return nil, nil, errors.New("Plugin strategy not set")
+	}
+
+	config := metricTypes[0].Config()
+	cfg := map[string]ctypes.ConfigValue{}
+	if config != nil {
+		cfg = config.Table()
+	}
+
+	pool.RLock()
+	defer pool.RUnlock()
+	p, serr := pool.SelectAP(taskID, cfg)
+	if serr != nil {
+		return nil, nil, serr
+	}
+
+	cli, ok := p.(*availablePlugin).client.(client.PluginStreamCollectorClient)
```

Comment:
Yes this is in StreamMetrics which is only implemented/intended for collectors.

We do verify that the assertion was successful.